mfsmaster/matocsserv.c

```diff
@@ -970,7 +970,9 @@ int matocsserv_get_csdata(void *e,uint32_t *servip,uint16_t *servport,uint32_t *
 	matocsserventry *eptr = (matocsserventry *)e;
 	if (eptr->mode!=KILL) {
 		*servip = eptr->servip;
-		*servport = eptr->servport;
+		if (servport!=NULL) {
```

Comment:
@acid-maker theres's no "matocsserv_server_get_ip" function, I modified matocsserv_get_csdata so that it can only return "servip", is that what you mean?

Reply:
My mistake. There is such function, but in 4.x (not yet released) :)
So leave it as is - anyway I'll need to adapt your code do 4.x, so I'll change it by myself during the process.